[RDY]Fixed ":input" screen refresh bug

motiejusvin opened the pull request:
Fixed bug: #14109

Commit:
Fixed ":input" screen refresh bug

/src/nvim/ex_cmds.c: added screen refresh

Commit:
Fixed ":input" screen refresh bug

src/nvim/ex_cmds.c: Fixed bug,added '{' to "else" statement

Commit:
Fixed formating errors

src/nvim/ex_cmds.c: added braces to "if" and "else if" statements

Commit:
Fixed :input

Commit:
Fixed ":input" screen refresh bug

src/nvim/ex_cmds.c: Added display refresh

bfredl commented on src/nvim/ex_cmds.c:
```diff
@@ -2861,12 +2861,14 @@ void ex_append(exarg_T *eap)
         vcol += 8 - vcol % 8;
-      else
+      } else {
+	update_screen(0);
```
we try to avoid direct calls to `update_screen` if possible. would `redraw_later(curwin, VALID)` work?

motiejusvin replied:
No. Screen does not redraw,if replaced `update_screen` to `redraw_later(curwin, VALID)`

teto replied:
buffer content changed, shouldn't it be NOT_VALID instead ?

bfredl replied:
That redraws everything which is wasteful. I would try `redrawWinline(wp, line)` first.

motiejusvin replied:
Fixed. Deleted `update_screen()` and inserted `redrawWinline(curwin,line)` to other place.

motiejusvin replied:
What about a for loop,that goes trough every line. Would that be okay in terms of resources?

motiejusvin replied:
Fixed. I inserted `redrawWinline(curwin,lnum)` a bit too far. Now its fixed using `redrawWinline(curwin,lnum)`

Commit:
Changed update_screen to redrawWinline

src/nvim/ex_cmds.c: update_screen deleted and inserted redrawWinline

Commit:
Changed location of redrawWinline and removed accidental commmit

/src/nvim/ex_cmds.c: changed location of redrawWinline and removed
accidental commit

Commit:
Fix formating

bfredl commented:
seems reasonable now :) maybe a test?

motiejusvin commented:
> seems reasonable now :) maybe a test?

I'm new to this open-source project. So by test it means `make test` ?

bfredl commented:
yes, `functionaltest` to be more specific. look for tests in `test/functional/ui/*_spec.lua` for examples of screen tests.

motiejusvin commented:
> yes, `functionaltest` to be more specific. look for tests in `test/functional/ui/*_spec.lua` for examples of screen tests.

Ran some tests. Functionals Without fix had 9 fails,and with had 4 fails.

Commit:
Removed unnecessary curly braces

src/nvim/ex_cmds.c: Last fix

Commit:
Removed white spaces

src/nvim/ex_cmds.c: removed whitespace

teto commented:
when you change code, the linter checks for the neighboring lines of your patch (right before and right after): it allows to refactor the codebase incrementally rather than just refactor everything at once.
see https://github.com/neovim/neovim/pull/14175/checks?check_run_id=2176786800#step:11:225 for the errors. 

Also the errors are diffed against a list of errors on master so it's a good idea to rebase usually.

Commit:
Merge remote-tracking branch 'upstream/master'

motiejusvin commented:
I guess everything's good now after updating the fork

Commit:
Trying to fix lint check

ex_cmds: lint check

teto commented:
when you address a linter issue, this changes the line and so the linter looks into the next neighbors, so you can end up changing many lines as it propagates.

motiejusvin commented:
So, if I understand correctly if I try to fix things, I will just make things worse. And, if I'm done with everything, I just change the title to [RDY]?

motiejusvin commented:
And thank you for all the time. I'll try to ask less in the future.
